### PR TITLE
Add MatchAll query

### PR DESCRIPTION
**Pull request by AsafMah**
Added MatchAll query.
This solves the issue when an error pops up when you just type "*"

### CODE REVIEW

File: K2Bridge.Tests.UnitTests/Visitors/LuceneNet/LuceneMatchAllDocsVisitorTests.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            var query = new LuceneMatchAllDocsQuery {
+                LuceneQuery = null,
+            };
+            var visitor = new LuceneVisitor();
+            Assert.That(
+                () => visitor.Visit(query),
+                Throws.TypeOf<IllegalClauseException>());
+        }
+
+        [TestCase(ExpectedResult = null)]
```
Comment (eladiw) [inline K2Bridge.Tests.UnitTests/Visitors/LuceneNet/LuceneMatchAllDocsVisitorTests.cs:38]: why expected result is null?

**Comment by eladiw** [comment]
looks good. one question

File: K2Bridge.Tests.UnitTests/Visitors/LuceneNet/LuceneMatchAllDocsVisitorTests.cs (hunk 1 above)
Comment (AsafMah) [reply K2Bridge.Tests.UnitTests/Visitors/LuceneNet/LuceneMatchAllDocsVisitorTests.cs:38]: See K2Bridge/Visitors/QueryStringClauseVisitor.cs.

Match all matches everything, so we don't need to add anything to the kusto query.
I can replace it with String.Empty if you prefer

**Review by eladiw — APPROVED** [review]
lgtm

**Comment by YannickBrombach** [comment]
lgtm

**Review by YannickBrombach — APPROVED** [review]
lgtm

**Pull request merged**